bin/improver-optical-flow

```diff
@@ -0,0 +1,141 @@
+                        help="Number of iterations to perform in enforcing "
+                        "smoothness constraint for optical flow velocities.")
+
+    # AdvectField options
+    parser.add_argument("--extrapolate", action="store_true", default=False,
+                        help="Optional flag to advect current data forward to "
+                        "specified lead times.")
+    parser.add_argument("--max_lead_time", type=int, default=360,
+                        help="Maximum lead time required (mins).  Ignored "
+                        "unless '--extrapolate' is set.")
+    parser.add_argument("--lead_time_interval", type=int, default=15,
+                        help="Interval between required lead times (mins). "
+                        "Ignored unless '--extrapolate' is set.")
+
+    args = parser.parse_args()
+
+    # read input data
+    cube_list = []
+    for fname in args.input_filepaths:
+        cube_list.append(load_cube(fname))
+    if len(cube_list) != 3:
+        raise ValueError("Expected exactly 3 input files.")
+
+    # order input files by validity time
+    cube_list.sort(key=lambda x: x.coord("time").points[0])
+    time_coord = cube_list[-1].coord("time")
+
+    # calculate optical flow velocities from T-1 to T and T-2 to T-1
+    ofc_plugin = OpticalFlow(iterations=args.smart_smoothing_iterations)
+    ucubes = iris.cube.CubeList([])
+    vcubes = iris.cube.CubeList([])
+    for older_cube, newer_cube in zip(cube_list[:-1], cube_list[1:]):
+        ucube, vcube = ofc_plugin.process(older_cube, newer_cube,
+                                          boxsize=args.ofc_box_size)
+        ucubes.append(ucube)
+        vcubes.append(vcube)
+
+    # average optical flow velocity components
+    ucube = ucubes.merge_cube()
+    umean = ucube.collapsed("time", iris.analysis.MEAN)
+    umean.coord("time").points = time_coord.points
+    umean.coord("time").units = time_coord.units
+
+    vcube = vcubes.merge_cube()
+    vmean = vcube.collapsed("time", iris.analysis.MEAN)
+    vmean.coord("time").points = time_coord.points
+    vmean.coord("time").units = time_coord.units
+
+    # save mean optical flow components as netcdf files
+    for wind_cube in [umean, vmean]:
+        file_name = generate_file_name(wind_cube)
+        save_netcdf(wind_cube, os.path.join(args.output_dir, file_name))
+
+    # advect latest input data to the required lead times
+    if args.extrapolate:
+
+        advection_plugin = AdvectField(umean, vmean)
+
+        # generate list of lead times in minutes
+        lead_times = np.arange(args.lead_time_interval, args.max_lead_time+1,
+                               args.lead_time_interval)
+
+        for lead_time in lead_times:
+            # cast to float as datetime.timedelta cannot accept np.int
+            timestep = datetime.timedelta(seconds=60.*lead_time)
+            forecast_cube = advection_plugin.process(cube_list[2], timestep)
```

Comment:
I think `cube_list[2]` could be replaced by `cube_list[-1]`, which would be a bit more flexible. Alternatively, a more descriptive name could be used.